aad/aad-b2c-with-spring-security/README.adoc

```diff
@@ -0,0 +1,228 @@
+= Tutorial of using Azure Active Directory B2C by Spring Security
+:toc:
+
+From this tutorial, you will learn how to use Azure Active Directory B2C by Spring Security.
+
+== Prerequisites
+- https://azure.microsoft.com/free[Azure Subscription]
+- https://docs.microsoft.com/java/azure/jdk/?view=azure-java-stable[Java Development Kit (JDK)] with version 8 or above
+
+[[create-azure-active-directory-B2C-resources]]
+
+== Prepare Azure Active Directory B2C resources
+
+=== Create your Azure Active Directory B2C tenant
+
+Follow the guide of https://docs.microsoft.com/azure/active-directory-b2c/tutorial-create-tenant[AAD B2C tenant creation].
+
+=== Register your Azure Active Directory B2C application
+
+Follow the guide of https://docs.microsoft.com/azure/active-directory-b2c/tutorial-register-applications[AAD B2C application registry].
+Please ensure that your b2c application's `Redirect URL` is configured to `http://localhost:8080/login/oauth2/code/`.
+In order to jump to the correct page (home) after `profile-edit` and `password-reset`, we need to add `http://localhost:8080/` in the redirect uri too.
+You can add additional `Redirect URL` with https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#add-a-redirect-uri[Add Redirect URL].
+
+=== Create user flows
+
+Follow the guide of https://docs.microsoft.com/azure/active-directory-b2c/tutorial-create-user-flows[AAD B2C user flows creation].
+
+=== Enable forgot password
+
+Follow the guide of https://docs.microsoft.com/en-us/azure/active-directory-b2c/add-password-reset-policy?pivots=b2c-user-flow#self-service-password-reset-recommended[AAD B2C enable forgot password].
+
+=== Enable profile edit
+
+Follow the guide of https://docs.microsoft.com/en-us/azure/active-directory-b2c/add-profile-editing-policy?pivots=b2c-user-flow[AAD B2C set up a profile edit flow].
+
+=== Enable password reset
+
+Follow the guide of https://docs.microsoft.com/en-us/azure/active-directory-b2c/add-password-reset-policy?pivots=b2c-user-flow#create-a-password-reset-user-flow[AAD B2C set up a password reset user flow]
+
+== Work as web application
+
+[[work-application-simple-sample]]
```

Comment:
Not all resource-servers samples are after web-application samples.
How about doing like this:

1. In this README.adoc, after `Prerequisites`, we just have contents this:
  * Sample 01: Web application
  * Sample 02: Web application validate audience
  * Sample 03: Web application authorize by scope
  * Sample 04: Web application authorize by app role
  * Sample 05: Web application authorize by client credential
  * Sample 06: Resource server
  * Sample 07: Web application access resource server
  * Sample 08: Resource server validate audience
  * Sample 09: Resource server authorize by scope
  * Sample 10: Resource server authorize by app role
  * Sample 11: Resource server multi tenant
2. Rename artifact to `sample-01-web-application`, `sample-02-web-application-validate-audence`...
3. Put `Prepare Azure Active Directory B2C resources` into sample-01, just before this:
```
Fill necessary values in application.yml, like ${your-jwk-set-uri}. 
Fill necessary values in home.html, like ${your-tenant-name}.
```
4. After above steps, please go through these samples one by one, then you may know how to improve this doc.